express-notes-api/server.js

```diff
@@ -49,7 +49,7 @@ app.post('/api/notes', async (req, res) => {
 app.delete('/api/notes/:id', async (req, res) => {
   try {
     const deleteId = req.params.id;
-    if (deleteId < 0) { // Checks to see if id is a positive integer //
+    if (deleteId < 0 || isNaN(Number(deleteId)) || Number(deleteId) === 0 || (Number(deleteId) % 1) !== 0) { // Checks to see if id is a positive integer //
```

Comment:
For safety, you should check `isNaN` first. That way if it fails the other checks won't execute.

You could also convert `deleteId` to Number when you first assign to it (on line 51).